src/EFCore.Relational/Extensions/RelationalQueryableExtensions.cs

```diff
@@ -158,12 +159,49 @@ public static DbCommand CreateDbCommand([NotNull] this IQueryable source)
                 Expression.Constant(arguments));
         }
 
+        /// <summary>
+        ///     <para>
+        ///         Returns a new query in which the collections in the query results will be loaded in single database query.
+        ///     </para>
+        ///     <para>
+        ///         This behavior ensures consistency of the data returned within the constraints of the transaction mode in use.
```

Comment:
It may be better to use stronger language. Unless the user specifically disabled SQL Server Read Committed Snapshot Isolation (RCSI) (or started a read uncommitted transaction), this should always ensure consistency. How about:

```suggestion
        ///         This behavior generally guarantees result consistency in the face of concurrent updates (but details may very based on the database and transaction isolation level in use).
```

Reply:
I don't understand it, so I will just update to whatever you say. 😉

Reply:
The actual details are unfortunately database-dependent... However, tl;dr:
* When EF Core creates SQL Server, it sets the RCSI mode which makes since single queries consistent. So everything is good unless you explicitly do stuff to opt out of it (disable RCSI, or start an uncommitted read transaction which is pure chaos)
* For PG and other MVCC-only databases (I think Oracle is in this bag) a single query is always consistent (because always operates on a point-in-time snapshot)

Not sure about Sqlite/MySQL. But the general advice is that single query should be safe by default.